### PR TITLE
filters/http: remove well_known_names

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Long Dai <long0dai@foxmail.com>

Commit Message:

Remove well_known_names in filters/http field, part of #7238

Additional Description:
Risk Level:
Risk Level: Low
Testing: functional
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: test/extensions/filters/http/common/fuzz/uber_per_filter.cc [hunk 1]
```diff
@@ -109,12 +108,12 @@ void UberFilterFuzzer::cleanFuzzedConfig(absl::string_view filter_name,
   const std::string name = Extensions::HttpFilters::Common::FilterNameUtil::canonicalFilterName(
       std::string(filter_name));
   // Map filter name to clean-up function.
-  if (filter_name == HttpFilterNames::get().GrpcJsonTranscoder) {
+  if (filter_name == "envoy.filters.http.grpc_json_transcoder") {
     // Add a valid service proto descriptor.
     addBookstoreProtoDescriptor(message);
-  } else if (name == HttpFilterNames::get().Squash) {
+  } else if (name == "envoy.filters.http.Squash") {
```
Comment (zuercher) [inline test/extensions/filters/http/common/fuzz/uber_per_filter.cc:114]: `Squash` should be all-lowercase.

File: test/extensions/filters/http/common/utility_test.cc [hunk 2]
```diff
@@ -35,9 +34,9 @@ TEST(FilterNameUtilTest, DEPRECATED_FEATURE_TEST(TestDeprecatedName)) {
       deprecatedFeatureEnabled("envoy.deprecated_features.allow_deprecated_extension_names", _))
       .WillRepeatedly(Return(true));
 
-  EXPECT_EQ(HttpFilterNames::get().Buffer,
+  EXPECT_EQ("envoy.filters.http.buffer",
             FilterNameUtil::canonicalFilterName("envoy.buffer", &runtime));
-  EXPECT_EQ(HttpFilterNames::get().Squash,
+  EXPECT_EQ("envoy.filters.http.Squash",
```
Comment (zuercher) [inline test/extensions/filters/http/common/utility_test.cc:39]: Same.

**Review by zuercher — COMMENTED** [review]
Thanks. Needs a merge and I noticed one error. Can you double check the others?

File: test/extensions/filters/http/common/utility_test.cc (hunk 2 above)
Comment (daixiang0) [reply test/extensions/filters/http/common/utility_test.cc]: thanks for review.

**Review by zuercher — APPROVED** [review]

**Pull request merged**